### ISSUE TITLE
fastlane cert --output_path doesn't generate p12 file

### ISSUE THREAD

**Issue by Sukhrob** [fastlane/fastlane#11009]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
    mkdir certificates
    fastlane cert --output_path certificates 

##### Complete output when running fastlane, including the stack trace and command used

Only `.cer` file is generated instead of three files `.cer`, `p12` and `certSigningRequest`. See https://circleci.com/docs/1.0/ios-code-signing/

I have to upload `p12` file to Circle CI which is missing.

### Environment

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13                                       |
| Ruby                        | 2.3.1                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.10.1                          |
| Installation Source         | /usr/local/bin/fastlane                     |
| Host                        | Mac OS X 10.13 (17A405)                     |
| Ruby Lib Dir                | /usr/local/Cellar/ruby/2.3.1_2/lib          |
| OpenSSL Version             | OpenSSL 1.0.2j  26 Sep 2016                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 8.3.2                                       |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.62.1  | 🚫 Update available |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                   | Version      |
| --------------------- | ------------ |
| did_you_mean          | 1.0.0        |
| slack-notifier        | 1.5.1        |
| rouge                 | 2.0.7        |
| xcpretty              | 0.2.8        |
| terminal-notifier     | 1.8.0        |
| unicode-display_width | 1.3.0        |
| terminal-table        | 1.8.0        |
| plist                 | 3.3.0        |
| public_suffix         | 2.0.5        |
| addressable           | 2.5.2        |
| multipart-post        | 2.0.0        |
| word_wrap             | 1.0.0        |
| babosa                | 1.0.2        |
| colored               | 1.2          |
| highline              | 1.7.8        |
| commander-fastlane    | 4.4.5        |
| excon                 | 0.59.0       |
| faraday               | 0.13.1       |
| unf_ext               | 0.0.7.4      |
| unf                   | 0.1.4        |
| domain_name           | 0.5.20170404 |
| http-cookie           | 1.0.3        |
| faraday-cookie_jar    | 0.0.6        |
| fastimage             | 2.1.0        |
| mini_magick           | 4.5.1        |
| multi_json            | 1.12.2       |
| multi_xml             | 0.6.0        |
| rubyzip               | 1.2.1        |
| security              | 0.1.3        |
| dotenv                | 2.2.1        |
| faraday_middleware    | 0.12.2       |
| uber                  | 0.1.0        |
| declarative           | 0.0.10       |
| declarative-option    | 0.1.0        |
| representable         | 3.0.4        |
| retriable             | 3.1.1        |
| mime-types-data       | 3.2016.0521  |
| mime-types            | 3.1          |
| httpclient            | 2.8.3        |
| google-api-client     | 0.13.6       |
| json                  | 2.1.0        |
| io-console            | 0.4.5        |
| tty-screen            | 0.5.1        |
| claide                | 1.0.2        |
| colored2              | 3.1.2        |
| nanaimo               | 0.2.3        |
| xcodeproj             | 1.5.3        |
| bundler               | 1.15.4       |
</details>

*generated on:* **2017-11-23**
</details>

**Comment by milch** [comment]
Does the `certificates` folder exist? If you run it without the `output_folder` parameter, does it work then?

**Comment by Sukhrob** [comment]
Sure. I created the folder first then ran the fastlane command.

**Comment by milch** [comment]
Could you post the full output of the command you ran? You can do this by appending `--capture-output` to your command, it'll create a template with everything we should need to get this sorted out

**Comment by Sukhrob** [comment]
It says `invalid option: --capture-output`

**Comment by milch** [comment]
Ah, sorry about that. The option is actually capture_output (with a _ instead of -) and it has to go between fastlane and cert

**Comment by Sukhrob** [comment]
### Captured Output

Command Used: `cert`
<details><summary>Output/Log</summary>

```


Looking for related GitHub issues on fastlane/fastlane...
Search query: Could not retrieve response as fastlane runs in non-interactive mode

URL: https://api.github.com/search/issues?q=Could%20not%20retrieve%20response%20as%20fastlane%20runs%20in%20non-interactive%20mode+repo:fastlane/fastlane
➡️  FR : Improve non interactive mode for latest_testflight_build_number
    https://github.com/fastlane/fastlane/issues/10772 [open] 3 💬
    3 weeks ago

➡️  [iOS] Alternative Solution to Could not retrieve response as fastlane runs in non-interactive mode 
    https://github.com/fastlane/fastlane/issues/10335 [closed] 8 💬
    4 days ago

➡️  Could not retrieve response as fastlane runs in non-interactive mode
    https://github.com/fastlane/fastlane/issues/9977 [closed] 6 💬
    5 weeks ago

and 23 more at: https://github.com/fastlane/fastlane/search?q=Could%20not%20retrieve%20response%20as%20fastlane%20runs%20in%20non-interactive%20mode&type=Issues&utf8=✓

🔗  You can ⌘ + double-click on links to open them directly in your browser.


```

</details>

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13                                       |
| Ruby                        | 2.3.1                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.10.1                          |
| Installation Source         | /usr/local/bin/fastlane                     |
| Host                        | Mac OS X 10.13 (17A405)                     |
| Ruby Lib Dir                | /usr/local/Cellar/ruby/2.3.1_2/lib          |
| OpenSSL Version             | OpenSSL 1.0.2j  26 Sep 2016                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 8.3.2                                       |


### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |


### fastlane files:

**No Fastfile found**


**No Appfile found**


### fastlane gems

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.62.1  | 🚫 Update available |


### Loaded fastlane plugins:

**No plugins Loaded**


<details><summary><b>Loaded gems</b></summary>

| Gem                   | Version      |
| --------------------- | ------------ |
| did_you_mean          | 1.0.0        |
| slack-notifier        | 1.5.1        |
| rouge                 | 2.0.7        |
| xcpretty              | 0.2.8        |
| terminal-notifier     | 1.8.0        |
| unicode-display_width | 1.3.0        |
| terminal-table        | 1.8.0        |
| plist                 | 3.3.0        |
| public_suffix         | 2.0.5        |
| addressable           | 2.5.2        |
| multipart-post        | 2.0.0        |
| word_wrap             | 1.0.0        |
| babosa                | 1.0.2        |
| colored               | 1.2          |
| highline              | 1.7.8        |
| commander-fastlane    | 4.4.5        |
| excon                 | 0.59.0       |
| faraday               | 0.13.1       |
| unf_ext               | 0.0.7.4      |
| unf                   | 0.1.4        |
| domain_name           | 0.5.20170404 |
| http-cookie           | 1.0.3        |
| faraday-cookie_jar    | 0.0.6        |
| fastimage             | 2.1.0        |
| mini_magick           | 4.5.1        |
| multi_json            | 1.12.2       |
| multi_xml             | 0.6.0        |
| rubyzip               | 1.2.1        |
| security              | 0.1.3        |
| dotenv                | 2.2.1        |
| faraday_middleware    | 0.12.2       |
| uber                  | 0.1.0        |
| declarative           | 0.0.10       |
| declarative-option    | 0.1.0        |
| representable         | 3.0.4        |
| retriable             | 3.1.1        |
| mime-types-data       | 3.2016.0521  |
| mime-types            | 3.1          |
| httpclient            | 2.8.3        |
| google-api-client     | 0.13.6       |
| json                  | 2.1.0        |
| io-console            | 0.4.5        |
| tty-screen            | 0.5.1        |
| claide                | 1.0.2        |
| colored2              | 3.1.2        |
| nanaimo               | 0.2.3        |
| xcodeproj             | 1.5.3        |
| bundler               | 1.15.4       |
| gh_inspector          | 1.0.3        |
</details>


*generated on:* **2017-11-23**
</details>

**Comment by Sukhrob** [comment]
Set environment variables and updated the fastlane to the latest version. But, the issue still exists.

### Captured Output

Command Used: `cert`
<details><summary>Output/Log</summary>

```


Looking for related GitHub issues on fastlane/fastlane...
Search query: Could not retrieve response as fastlane runs in non-interactive mode

URL: https://api.github.com/search/issues?q=Could%20not%20retrieve%20response%20as%20fastlane%20runs%20in%20non-interactive%20mode+repo:fastlane/fastlane
➡️  FR : Improve non interactive mode for latest_testflight_build_number
    https://github.com/fastlane/fastlane/issues/10772 [open] 3 💬
    3 weeks ago

➡️  [iOS] Alternative Solution to Could not retrieve response as fastlane runs in non-interactive mode 
    https://github.com/fastlane/fastlane/issues/10335 [closed] 8 💬
    4 days ago

➡️  fastlane cert --output_path doesn't generate p12 file
    https://github.com/fastlane/fastlane/issues/11009 [open] 8 💬
    8 minutes ago

and 24 more at: https://github.com/fastlane/fastlane/search?q=Could%20not%20retrieve%20response%20as%20fastlane%20runs%20in%20non-interactive%20mode&type=Issues&utf8=✓

🔗  You can ⌘ + double-click on links to open them directly in your browser.


```

</details>

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13                                       |
| Ruby                        | 2.3.1                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.10.1                          |
| Installation Source         | /usr/local/bin/fastlane                     |
| Host                        | Mac OS X 10.13 (17A405)                     |
| Ruby Lib Dir                | /usr/local/Cellar/ruby/2.3.1_2/lib          |
| OpenSSL Version             | OpenSSL 1.0.2j  26 Sep 2016                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 8.3.2                                       |


### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |


### fastlane files:

**No Fastfile found**


**No Appfile found**


### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.66.2  | ✅ Up-To-Date  |


### Loaded fastlane plugins:

**No plugins Loaded**


<details><summary><b>Loaded gems</b></summary>

| Gem                   | Version      |
| --------------------- | ------------ |
| did_you_mean          | 1.0.0        |
| slack-notifier        | 1.5.1        |
| claide                | 1.0.2        |
| colored2              | 3.1.2        |
| nanaimo               | 0.2.3        |
| xcodeproj             | 1.5.3        |
| rouge                 | 2.0.7        |
| xcpretty              | 0.2.8        |
| terminal-notifier     | 1.8.0        |
| unicode-display_width | 1.3.0        |
| terminal-table        | 1.8.0        |
| plist                 | 3.3.0        |
| public_suffix         | 2.0.5        |
| addressable           | 2.5.2        |
| multipart-post        | 2.0.0        |
| word_wrap             | 1.0.0        |
| tty-screen            | 0.6.3        |
| babosa                | 1.0.2        |
| colored               | 1.2          |
| highline              | 1.7.8        |
| commander-fastlane    | 4.4.5        |
| excon                 | 0.59.0       |
| faraday               | 0.13.1       |
| unf_ext               | 0.0.7.4      |
| unf                   | 0.1.4        |
| domain_name           | 0.5.20170404 |
| http-cookie           | 1.0.3        |
| faraday-cookie_jar    | 0.0.6        |
| fastimage             | 2.1.0        |
| mini_magick           | 4.5.1        |
| multi_json            | 1.12.2       |
| multi_xml             | 0.6.0        |
| rubyzip               | 1.2.1        |
| security              | 0.1.3        |
| dotenv                | 2.2.1        |
| faraday_middleware    | 0.12.2       |
| uber                  | 0.1.0        |
| declarative           | 0.0.10       |
| declarative-option    | 0.1.0        |
| representable         | 3.0.4        |
| retriable             | 3.1.1        |
| mime-types-data       | 3.2016.0521  |
| mime-types            | 3.1          |
| httpclient            | 2.8.3        |
| google-api-client     | 0.13.6       |
| json                  | 2.1.0        |
| io-console            | 0.4.5        |
| bundler               | 1.15.4       |
| gh_inspector          | 1.0.3        |
</details>


*generated on:* **2017-11-23**
</details>

**Comment by milch** [comment]
Alright, if you didn't truncate anything from the log it looks like you might be running straight `cert` without any of the parameters, somewhere where you can't supply any additional inputs (e.g. you're running this command on CI), because your error message seems to be `Could not retrieve response as fastlane runs in non-interactive mode`. You can try running fastlane somewhere else, where you could supply input for it, or you can check the supported parameters by running `fastlane action cert`

**Comment by krosti** [comment]
I'm having the same issue. Any news on this?

**Comment by akamud** [comment]
I can confirm this is happening to me too. Fastlane 2.71.1.

**Comment by krazik** [comment]
+1 seeing this

**Comment by krosti** [comment]
I figured out that using 

`fastlane cert --output_path certificates --development true --force true --platform ios ` 

works 👍

**Comment by akamud** [comment]
This is fixed in version 2.75.1+
